source/common/common/radix_tree.h

```diff
@@ -0,0 +1,324 @@
+      }
+
+      // Look for the edge
+      uint8_t firstChar = static_cast<uint8_t>(search[0]);
+      auto childIt = children_.find(firstChar);
+
+      // No edge, create one
+      if (childIt == children_.end()) {
+        // Create a new child node
+        RadixTreeNode newChild;
+        newChild.prefix_ = std::string(search);
+        newChild.value_ = std::move(value);
+
+        // Add the child to the current node
+        children_[firstChar] = std::move(newChild);
+        return;
+      }
+
+      // Get the child node
+      RadixTreeNode& child = childIt->second;
+
+      // Determine longest prefix of the search key on match
+      size_t commonPrefix = longestPrefix(search, child.prefix_);
+      if (commonPrefix == child.prefix_.size()) {
+        // The search key is longer than the child prefix, continue down
+        absl::string_view remainingSearch = search.substr(commonPrefix);
+        child.insert(key, remainingSearch, std::move(value));
+        return;
+      }
+
+      // Split the node - create a new intermediate node
+      RadixTreeNode splitNode;
+      splitNode.prefix_ = std::string(search.substr(0, commonPrefix));
+
+      // Update the child's prefix
+      child.prefix_ = std::string(child.prefix_.substr(commonPrefix));
+
+      // If the search key is exactly the common prefix, set the value on the split node
+      if (commonPrefix == search.size()) {
+        splitNode.value_ = std::move(value);
+      } else {
+        // Create a new leaf for the current key
+        RadixTreeNode newLeaf;
+        newLeaf.prefix_ = std::string(search.substr(commonPrefix));
+        newLeaf.value_ = std::move(value);
+        splitNode.children_[static_cast<uint8_t>(newLeaf.prefix_[0])] = std::move(newLeaf);
+      }
+
+      // Add the child to the split node
+      splitNode.children_[static_cast<uint8_t>(child.prefix_[0])] = std::move(child);
+
+      // Replace the original child with the split node
+      children_[firstChar] = std::move(splitNode);
+    }
+
+    /**
+     * Recursive helper for find operation.
+     * @param search the remaining search key.
+     * @param result the value to return if found.
+     * @return true if the key was found, false otherwise.
+     */
+    bool find_recursive(absl::string_view search, Value& result) const {
+      if (search.empty()) {
+        if (hasValue(*this)) {
+          result = value_;
+          return true;
+        }
+        return false;
+      }
+
+      uint8_t firstChar = static_cast<uint8_t>(search[0]);
+      auto childIt = children_.find(firstChar);
+      if (childIt == children_.end()) {
+        return false;
+      }
+
+      const RadixTreeNode& child = childIt->second;
+
+      // Check if the child's prefix matches the search
+      if (search.size() >= child.prefix_.size() &&
+          search.substr(0, child.prefix_.size()) == child.prefix_) {
+        absl::string_view newSearch(search.begin() + child.prefix_.size(), search.end());
```

Comment:
```
absl::string_view newSearch = search.substr(child.prefix_.size());
```
For readability and consistency with how this is done every other time.

Or just roll `search.substr` directly in the next line and don't have an intermediate temp variable at all.